### PR TITLE
[flang][openacc] Place post allocate/deallocate attribute correctly

### PR DESCRIPTION
**Pull request by clementval**
The `acc.declate_action` attribute was sometime misplaced as reported in #79770.
This patch updates the lowering code to place the postAllocate/postDeallocate actions at the correct place.

### CODE REVIEW

File: flang/test/Lower/OpenACC/acc-declare.f90 [hunk 1]
```diff
@@ -318,6 +318,18 @@ subroutine acc_declare_array_section(a)
...
+    real, pointer, dimension(:) :: localptr
+    !$acc declare create(localptr)
+    allocate(localptr(n), stat=status)
+
+    deallocate(localptr, stat=status)
+  end subroutine
+
+! CHECK-LABEL: func.func @_QMacc_declarePacc_declare_allocate_with_stat()
+! CHECK: fir.call @_FortranAPointerAllocate(%{{.*}}, %{{.*}}, %{{.*}}, %{{.*}}, %{{.*}}) {{.*}} {acc.declare_action = #acc.declare_action<postAlloc = @_QMacc_declareFacc_declare_allocate_with_statElocalptr_acc_declare_update_desc_post_alloc>}
+! CHECK: fir.call @_FortranAPointerDeallocate(%{{.*}}, %{{.*}}, %{{.*}}, %{{.*}}, %{{.*}}) {{.*}} {acc.declare_action = #acc.declare_action<preDealloc = @_QMacc_declareFacc_declare_allocate_with_statElocalptr_acc_declare_update_desc_pre_dealloc>}
```
Comment (vzakhari) [inline flang/test/Lower/OpenACC/acc-declare.f90:332]: Is it worth checking for proper attachment of the post-deallocate action?

**Review by vzakhari — APPROVED** [review]
Thank you for the fix!

File: flang/test/Lower/OpenACC/acc-declare.f90 (hunk 1 above)
Comment (clementval) [reply flang/test/Lower/OpenACC/acc-declare.f90:332]: I'll add it.

Comment (clementval) [reply flang/test/Lower/OpenACC/acc-declare.f90]: It was indeed good to check because it was actually overwritten by the preDeallocAction.

**Review by vzakhari — APPROVED** [review]
Thanks for fixing the attribute overwrite!

**Pull request merged**